client/chat/chat-reducer.js

```diff
@@ -119,6 +119,7 @@ export default keyedReducer(new ChatState(), {
       return m.push(
         new SelfJoinChannelMessageRecord({
           id: cuid(),
+          time: Date.now(),
```

Comment:
In general this not a thing you want to do, as reducers are meant to be pure and deterministic. This time needs to be derived from something in the action (whether that's a time value passed by the action creator, or from some other part of the init message that already exists). I don't really think this is safe to do with a client-side `Date.now()` value either way, as all the other times in messages are server-provided. Ideally the server would provide a time value for this as well.

Reply:
Fair enough. But I thought about fixing this in a separate PR, as a part of this issue: https://github.com/ShieldBattery/ShieldBattery/issues/139. Because as far as I can tell, we're doing this same thing for all the other non-user messages. I was thinking of moving the chat stuff to use HTTP pretty soon, and then do some server-side cleanup of chat stuff as a part of that.